From dc80078b471873f67f5cbb5d43309869825a788c Mon Sep 17 00:00:00 2001
From: Zbigniew Bodek <zbb@semihalf.com>
Date: Wed, 25 Jul 2012 12:52:46 +0200
Subject: [PATCH 0400/1825] DSMP, Fix compil. Delete vmalloc.h and add
 VMALLOC_END definition to pgtable.h

https://github.com/MISL-EBU-System-SW/misl-windriver.git linux-3.4.69-14t2-read
commit ec336adc8be7f6bc3ba519554f4c6306a28b5f68

- vmalloc.h is deprecated
- define VMALLOC_END for ARMADA_XP in pgtable.h

Change-Id: I327526c1cab650aa883ac89c191e6b9a99b97468
Signed-off-by: Kosta Zertsekel <konszert@marvell.com>
Signed-off-by: Zhong Hongbo <hongbo.zhong@windriver.com>
---
 arch/arm/include/asm/pgtable.h                |    6 ++++++
 arch/arm/mach-armadaxp/include/mach/vmalloc.h |   11 -----------
 2 files changed, 6 insertions(+), 11 deletions(-)
 delete mode 100644 arch/arm/mach-armadaxp/include/mach/vmalloc.h

diff --git a/arch/arm/include/asm/pgtable.h b/arch/arm/include/asm/pgtable.h
index bea0c61..39aef5c 100644
--- a/arch/arm/include/asm/pgtable.h
+++ b/arch/arm/include/asm/pgtable.h
@@ -42,6 +42,12 @@
 #define VMALLOC_START		(((unsigned long)high_memory + VMALLOC_OFFSET) & ~(VMALLOC_OFFSET-1))
 #ifdef CONFIG_ARCH_MX6Q
 #define VMALLOC_END		0xfb000000UL
+#elif defined(CONFIG_ARMADA_XP)
+/* Dove LCD driver performs big allocations for FrameBuffer memory, we need to
+ * move CONSISTENT_BASE by 32MB
+ */
+/* Was 0x2000000 */
+#define VMALLOC_END		0xfa800000UL
 #else
 #define VMALLOC_END		0xff000000UL
 #endif
diff --git a/arch/arm/mach-armadaxp/include/mach/vmalloc.h b/arch/arm/mach-armadaxp/include/mach/vmalloc.h
deleted file mode 100644
index 1630fe6..0000000
--- a/arch/arm/mach-armadaxp/include/mach/vmalloc.h
+++ /dev/null
@@ -1,11 +0,0 @@
-/*
- * include/asm-arm/arch-aurora/vmalloc.h
- */
-
-/* Dove LCD driver performs big allocations for FrameBuffer memory, we need to
- * move CONSISTENT_BASE by 32MB
- */
-/* Was 0x2000000 */
-
-#define VMALLOC_END	(0xfa800000)
-
-- 
1.7.5.4

